core/services/pipeline/orm.go

```diff
@@ -159,6 +157,47 @@ func (o *orm) CreateRun(ctx context.Context, jobID int32, meta map[string]interf
             SELECT ? AS pipeline_run_id, id AS pipeline_task_spec_id, type, index, NOW() AS created_at
             FROM pipeline_task_specs
             WHERE pipeline_spec_id = ?`, run.ID, run.PipelineSpecID).Error
+		if err != nil {
+			return err
+		}
+
+		// Ammend the TaskRuns with their predecessors.
+		var tr []TaskRun
```

Comment:
Can we save a query by using `RETURNING` in the INSERT and scanning into the TaskRun array?